peekingduck/pipeline/nodes/draw/poses.py

```diff
@@ -60,5 +103,75 @@ def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
             inputs (dict): Dictionary with keys "img", "keypoints", and
                 "keypoint_conns".
         """
-        draw_human_poses(inputs["img"], inputs["keypoints"], inputs["keypoint_conns"])
+        self.pose.draw_human_poses(
+            inputs["img"],
+            inputs["keypoints"],
+            inputs["keypoint_conns"],
+        )
         return {}
+
+    def _check_valid_color(self, key: str) -> None:
+        """Checks that configuration value specified by `key` is a valid color.
+
+        Example:
+            >>> keypoint_dot_color = "blueee"
+            >>> self._check_valid_color("keypoint_dot_color")
+
+            This will raise a ValueError as "blueee" is not a valid color.
+
+            >>> keypoint_dot_color = [100, 100, 300]
+            >>> self._check_valid_color("keypoint_dot_color")
+
+            This will raise a ValueError as 300 is out of range.
+
+        Args:
+            key (str): The specified key.
+
+        Raises:
+            ValueError: If the configuration value is not either a valid color
+                in :ref:`color-palette`, or a list of BGR values, or if the BGR
+                values are out of range [0, 255].
+        """
+        valid_color_names = set(COLOR_MAP.keys())
+        valid_color_range = "[0, 255]"
+
+        if isinstance(self.config[key], str):
+            try:
+                self.check_valid_choice(key, valid_color_names)  # type: ignore
+            except Exception as wrong_color_choice:
+                raise ValueError(
+                    f"{key} must be one of {sorted(valid_color_names)} or passed as a \
+                    list of BGR values."
+                ) from wrong_color_choice
+        else:
+            self._check_valid_length(key)
+            self.check_bounds(key, valid_color_range)
```

Comment:
Check valid length seems vague here, and doesn't really inform us what we consider to be a valid length.

Since we already know we're dealing with a list contain BGR value by the time we reach `else`, consider doing both the length and individual color channel value checks in a `_check_valid_bgr_value()` method. Example:

```
def _check_valid_bgr_value(self, key, color_range):
    if len(self.config[key]) != 3:
        raise ...
    self.check_bounds(key, valid_color_range)
```
I think it's fine to use a magic number `3` there since we already specify in the method name that we're checking BGR values

Reply:
Changed.